src/Rules/ConsoleCommand/NoConstructorDependencyInjectionRule.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Larastan\Larastan\Rules\ConsoleCommand;
+
+use Illuminate\Console\Command;
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InClassMethodNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+
+use function count;
+use function sprintf;
+
+/** @implements Rule<InClassMethodNode> */
+final class NoConstructorDependencyInjectionRule implements Rule
+{
+    public function getNodeType(): string
+    {
+        return InClassMethodNode::class;
+    }
+
+    /**
+     * @param InClassMethodNode $node
+     *
+     * @return RuleError[] errors
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        $method = $node->getMethodReflection();
+
+        if (! $method->isConstructor()) {
+            return [];
+        }
+
+        $classReflection = $node->getClassReflection();
+
+        if (! $classReflection->isSubclassOf(Command::class)) {
+            return [];
+        }
+
+        if ($classReflection->isAbstract()) {
+            return [];
+        }
+
+        $methodNode = $node->getOriginalNode();
+
+        if (count($methodNode->params) > 0) {
+            return [
+                RuleErrorBuilder::message(
+                    sprintf(
+                        'Console command "%s" should not have constructor arguments. Use dependency injection in the handle() or __invoke() method instead.',
+                        $classReflection->getName(),
+                    ),
+                )->line($methodNode->getLine())
+                    ->tip('Move all dependencies to the handle() or __invoke() method parameters for better testability and Laravel best practices.')
+                    ->identifier('larastan.console.constructorInjection')
```

Comment:
generally, I've only ever seen two components to the identifier, I would recommend:


```suggestion
                    ->identifier('larastan.consoleConstructorInjection')
```

Reply:
there are a bunch:

- src/Rules/DeferrableServiceProviderMissingProvidesRule.php:55
- src/Rules/UselessConstructs/NoUselessValueFunctionCallsRule.php:51

and more, but the most important one is the:
- src/Rules/ConsoleCommand/UndefinedArgumentOrOptionRule.php:78

which resides at the same namespace level as the newly proposed rule.
for the sake of continuety, I would suggest to leave this one out.

wdty ?

Reply:
I would still say those are the exception and not the rule, but it's up to you

Reply:
Then I am performing the change accordingly and I will create a corresponding issue with "help wanted"

Reply:
@calebdw https://github.com/larastan/larastan/pull/2324/commits/5874cac31b32fb54f0aed5dc8ee0f3dbf9f86757